docs/src/proposals/vote_subcommittee.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Vote Subcommittee
+---
+
+## Problem
+
+Each additional validator adds 1 vote per slot, increasing the
+message and compute load on the network.
+
+## Solution Overview
+
+Under the assumption that Turbine is able to guarantee that all
+honest nodes receive the block as long as at least X honest nodes
+receive the block, it is possible to allow a sampled set of validators
+to vote on that block as apposed to all the validators, and achieve
+the same liveness properties. If the network is tolerant of 1/3
+failures, then for a large enough random sample, like 200 or more,
+the probability of containing 2/3+ faulty nodes is 1:10^40. Assuming
+at least 1/3+ honest nodes are present in the subcommittee, then
+faulty forks will not get finality, and network will be able to
+continue.
+
+## Detailed Solution
+
+The following sections provide more details of the design.
+
+### Definitions
+
+* Voting Subcommittee: the set of nodes currently voting on blocks
+
+* Voting epoch: the number of slots that the voting subcommittee
+is voting for
+
+* primary subcommittee: The half of the voting subcommittee that
+is scheduled for its second epoch.
+
+* secondary subcommittee: The half of the voting subcommittee that
+is scheduled for its first epoch.
+
+* rotation block: The last block in the previous epoch that was
+rooted by 2/3+ of both primary and secondary super-majorities.
+
+* rotation seed: The seed used to generate the random sample of nodes.
+It's computed as `slow_hash(rotation block bank_hash)`
+
+
+* `slow_hash`: repeated sha256 for rounds equal to 1ms on modern
+sha-ni hardware.
+
+### Safety Violations
+
+An Optimistic Confirmation violation occurs if a voting subcommittee
+optimistically confirms a slot that wasn't rooted by the following
+voting subcommittee.
+
+A Safety violation occurs if a subcommittee roots a slot that wasn't
+rooted by the following subcommittee.
+
+All full nodes including RPCs should halt as soon as a safety
+violation is detected.
+
+### Subcommittee Rotation
+
+Voting Subcommittee is scheduled for a voting epoch, and is itself
+composed of 2 subcommittees, primary and secondary, scheduled for
+two epochs each in a staggered rotation.
+
+```
+A1 A1 A2 A2 A3
+   B1 B1 B2 B2
+```
+
+The first epoch the subcommittee runs, it is considered **secondary**.
+On the second epoch, it is considered **primary**.
+
+For a new secondary subcommittee to start, it must start with a
+block that is a descendant of a **rotation block** that was confirmed
```

Comment:
How long are voting epoch's? What if there is no rotation block - previous epoch ended with no roots.

Reply:
I think they can be as short as 32 slots, if all succeed. But I think it’s probably better to run it at 128.

Reply:
If there is no rooted block, then the committee doesn’t change. It continues as (A2, B1) for another epoch.

Reply:
B1 continues to be primary if there is no rooted slot, although might work if it rotates as well.  maybe thats a v1 not a v0 thing.